### PR TITLE
Added clone/restore state methods to ale_python_interface.

### PR DESCRIPTION
**Pull request by mhauskn**
Tested that clone/restore works correctly on Ubuntu. Not sure how to verify the integrity of RNG serialization, but if it's working in the ALE_inteface, it'll be working in the python as well.

### CODE REVIEW

File: ale_python_interface/ale_python_interface.py [hunk 1]
```diff
@@ -181,13 +190,46 @@ def getRAM(self, ram=None):
...
+        outcomes.  By contrast, see restoreSystemState.
+        """
+        ale_lib.restoreState(self.obj, state)
+
+    def cloneSystemState(self):
+        """This makes a copy of the system & environment state, suitable for
+        serialization. This includes pseudorandomness and so is *not*
+        suitable for planning purposes.
+        """
+        return ale_lib.cloneState(self.obj)
```
Comment (mgbellemare) [inline ale_python_interface/ale_python_interface.py]: Do you mean to call cloneSystemState here?

File: ale_python_interface/ale_python_interface.py [hunk 2]
```diff
@@ -181,13 +190,46 @@ def getRAM(self, ram=None):
...
+
+    def restoreSystemState(self, state):
+        """Reverse operation of cloneSystemState."""
+        ale_lib.restoreState(self.obj, state)
```
Comment (mgbellemare) [inline ale_python_interface/ale_python_interface.py]: restoreSystemState?

**Comment by mhauskn** [comment]
Good catches :+1:  I've fixed and tested the (clone/restore)SystemState calls. They seem to work fine, but I didn't verify that RNG is correctly cloned/restored.

**Comment by mgbellemare** [comment]
It is in C++. I'll re-verify before merging dev-rng-serialization-fix.

**Pull request merged**